core/io/resource.cpp

```diff
@@ -390,22 +390,9 @@ void Resource::_take_over_path(const String &p_path) {
 }
 
 RID Resource::get_rid() const {
-	if (get_script_instance()) {
-		Callable::CallError ce;
-		RID ret = get_script_instance()->callp(SNAME("_get_rid"), nullptr, 0, ce);
-		if (ce.error == Callable::CallError::CALL_OK && ret.is_valid()) {
-			return ret;
-		}
-	}
-	if (_get_extension() && _get_extension()->get_rid) {
-		RID ret;
-		ret.from_uint64(_get_extension()->get_rid(_get_extension_instance()));
-		if (ret.is_valid()) {
-			return ret;
-		}
-	}
```

Comment:
Applies suggested change and wrapped it in `DISABLE_DEPRECATED`.

Reply:
@dsnopek Do we also need to remove it from [`gdextension_interface.h`](https://github.com/godotengine/godot/blob/3978628c6cc1227250fc6ed45c8d854d24c30c30/core/extension/gdextension_interface.h#L360) with a new `GDExtensionClassCreationInfo4`?

Reply:
> @dsnopek Do we also need to remove it from [`gdextension_interface.h`](https://github.com/godotengine/godot/blob/3978628c6cc1227250fc6ed45c8d854d24c30c30/core/extension/gdextension_interface.h#L360) with a new `GDExtensionClassCreationInfo4`?

It's not required, but it would be good, in order to prevent confusion for future GDExtension binding implementors.

Reply:
Should I simply remove that line or wrap in DISABLE_DEPRECATED?

Reply:
@KoBeWi If you're referring to changes to `gdextension_interface.h`, you can't remove anything from the old struct, you'd have to make a new struct without that field.